ngraph/src/ngraph/runtime/reference/prior_box.hpp

```diff
@@ -0,0 +1,293 @@
+                                                                  shift / 2.f + c * shift;
+                                            float center_y_temp = center_y - fixed_size_ / 2 +
+                                                                  shift / 2.f + r * shift;
+                                            calculate_data(center_x_temp,
+                                                           center_y_temp,
+                                                           box_width_ratio,
+                                                           box_height_ratio,
+                                                           true);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+
+                        for (size_t ms_idx = 0; ms_idx < min_size.size(); ms_idx++)
+                        {
+                            box_width = min_size[ms_idx] * 0.5f;
+                            box_height = min_size[ms_idx] * 0.5f;
+                            calculate_data(center_x, center_y, box_width, box_height, false);
+
+                            if (attrs.max_size.size() > ms_idx)
+                            {
+                                box_width = box_height =
+                                    std::sqrt(min_size[ms_idx] * attrs.max_size[ms_idx]) * 0.5f;
+                                calculate_data(center_x, center_y, box_width, box_height, false);
+                            }
+
+                            if (attrs.scale_all_sizes ||
+                                (!attrs.scale_all_sizes && (ms_idx == min_size.size() - 1)))
+                            {
+                                size_t sIdx = attrs.scale_all_sizes ? ms_idx : 0;
+                                for (float ar : aspect_ratios)
+                                {
+                                    if (std::fabs(ar - 1.0f) < 1e-6)
+                                    {
+                                        continue;
+                                    }
+
+                                    ar = std::sqrt(ar);
+                                    box_width = min_size[sIdx] * 0.5f * ar;
+                                    box_height = min_size[sIdx] * 0.5f / ar;
+                                    calculate_data(
+                                        center_x, center_y, box_width, box_height, false);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if (attrs.clip)
+                {
+                    for (size_t i = 0; i < H * W * num_priors * 4; ++i)
+                    {
+                        dst_data[i] = (std::min)((std::max)(dst_data[i], 0.0f), 1.0f);
+                    }
+                }
+
+                size_t channel_size = OH * OW;
```

Comment:
You can get integer overflow. `size_t = int_64_t * int64_t`

Moreover do we really need this? As I can see OW is always equal to 1.
`const int64_t OW = 1;`